pandas/tests/tslibs/test_timedeltas.py

```diff
@@ -34,3 +34,41 @@ def test_huge_nanoseconds_overflow():
     # GH 32402
     assert delta_to_nanoseconds(Timedelta(1e10)) == 1e10
     assert delta_to_nanoseconds(Timedelta(nanoseconds=1e10)) == 1e10
+
+
+class TestComputingDeltasWithTimezoneDifferences:
+    def test_fails_to_subtract_when_one_date_is_naive(self):
+        time1 = Timestamp("2020-10-28 22:08:00-04:00")
+        time2 = Timestamp("2020-10-28 22:08:00")
+
+        msg = (
+            "Cannot compute time difference between "
+            "timezone-aware and naive datetime values."
+        )
+        with pytest.raises(TypeError, match=msg):
+            time1 - time2
+
+        with pytest.raises(TypeError, match=msg):
+            time2 - time1
+
+    def test_subtracts_dates_of_different_timezones(self):
+        time1 = Timestamp("2020-10-28 22:08:00-04:00")
+        time2 = Timestamp("2020-10-28 22:08:00-00:00")
+
+        assert time1 - time2 == Timedelta("0 days 04:00:00")
+
+    def test_subtracts_dates_of_different_types(self):
+        """ Github issue 31793 """
+
+        time1 = Timestamp("2020-10-28 22:08:00-04:00")
+        time2 = Timestamp("2020-10-28 22:08:00-00:00").to_pydatetime()
```

Comment:
maybe just use the `datetime.datetime` constructor

Reply:
ok, i am moving the tests to `tests/arithmetic/test_timedelta64.py` since most of this functionality is already covered here